pkg/kubelet/kubelet.go

```diff
@@ -2867,6 +2867,7 @@ func (kl *Kubelet) syncNetworkStatus() {
 
 // Set addresses for the node.
 func (kl *Kubelet) setNodeAddress(node *api.Node) error {
+	hostnameAddress := api.NodeAddress{Type: api.NodeHostName, Address: kl.GetHostname()}
```

Comment:
I think it would be helpful to map out all the combinations (with and without cloud provider, with and without hostname override, with and without the hostname being resolvable by the master), and show when the hostname override is honored and who is responsible for setting it. I also wonder if some cloud providers need to be updated to set this info if they are known to determine hostnames that won't resolve